api/opds2/publication.py

```diff
@@ -165,6 +165,16 @@ def parseEditionToPublication(self, editionRecord):
         # Covers
         self.findAndAddCover(editionRecord)
 
+        host = 'digital-research-books-beta'\
+        if os.environ['ENVIRONMENT'] == 'production' else 'drb-qa'
+
+        # Read Online Link
+        self.addLink({
+                'href': 'https://{}.nypl.org/read/{}'.format(host, editionRecord.id),
+                'rel': 'alternate',
+                'type': 'application/pdf'
```

Comment:
I'd look down at L182-184 for what this should be. The `/read/{}` path should contain a `link.id` not an `edition.id`. And it should only be done when we have a flag of `readable:True`. So I think it needs to be done in that loop and have some kind of conditional for what kind of object should be passed to `self.addLink`